src/Microsoft.AspNet.OData.Shared/DefaultEdmPatchMethodHandler.cs

```diff
@@ -0,0 +1,165 @@
+        {
+            createdObject = null;
+            errorMessage = string.Empty;
+
+            try
+            {
+                createdObject = new EdmEntityObject(entityType);
+                originalList.Add(createdObject);
+
+                return ODataAPIResponseStatus.Success;
+            }
+            catch (Exception ex)
+            {
+                errorMessage = ex.Message;
+
+                return ODataAPIResponseStatus.Failure;
+            }
+        }
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes")]
+        public override ODataAPIResponseStatus TryDelete(IDictionary<string, object> keyValues, out string errorMessage)
+        {
+            errorMessage = string.Empty;
+
+            try
+            {
+                EdmStructuredObject originalObject = GetFilteredItem(keyValues);
+                
+                if (originalObject != null)
+                {
+                    originalList.Remove(originalObject);
+                }
+
+                return ODataAPIResponseStatus.Success;
+            }
+            catch (Exception ex)
+            {
+                errorMessage = ex.Message;
+
+                return ODataAPIResponseStatus.Failure;
+            }
+        }
+
+        public override EdmODataAPIHandler GetNestedHandler(IEdmStructuredObject parent, string navigationPropertyName)
+        {
+            IEdmNavigationProperty navProperty = entityType.NavigationProperties().FirstOrDefault(navProp => navProp.Name == navigationPropertyName);
+
+            if(navProperty == null)
+            {
+                return null;
+            }
+
+            IEdmEntityType nestedEntityType = navProperty.ToEntityType();
+
+            object obj;
+            if(parent.TryGetPropertyValue(navigationPropertyName, out obj))
+            {
+                ICollection<IEdmStructuredObject> nestedList = obj as ICollection<IEdmStructuredObject>;
+
+                return new DefaultEdmODataAPIHandler(nestedList, nestedEntityType);
+            }            
+
+            return null;
+        }
+
+
+        private EdmStructuredObject GetFilteredItem(IDictionary<string, object> keyValues)
+        {
+            //This logic is for filtering the object based on the set of keys,
+            //There will only be very few key elements usually, mostly 1, so performance wont be impacted.
+
+            if(originalList == null)
+            {
+                return null;
+            }
+
+            foreach (EdmStructuredObject item in originalList)
+            {
+                bool isMatch = true;
+
+                foreach (KeyValuePair<string, object> keyValue in keyValues)
+                {
+                    object value;
+                    if (item.TryGetPropertyValue(keyValue.Key, out value))
+                    {
+                        if (!Equals(value, keyValue.Value))
```

Comment:
Wondering why we can't reverse the check here `if (Equals(value, keyValue.Value)` and return the item?